parsons/rockthevote/rtv.py

```diff
@@ -0,0 +1,211 @@
+import datetime
+import logging
+import petl
+import re
+import requests
+import time
+
+from parsons import Table
+from parsons.utilities import check_env
+from parsons.utilities.api_connector import APIConnector
+
+logger = logging.getLogger(__name__)
+
+DATETIME_FORMAT = '%Y-%m-%d %H:%M:%S UTC'
+"""Datetime format for sending date's to the API."""
+
+REQUEST_HEADERS = {
+    # For some reason, RTV's firewall REALLY doesn't like the
+    # user-agent string that Python's request library gives by default,
+    # though it seems fine with the curl user agent
+    # For more info on user agents, see:
+    # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent
+    'user-agent': 'curl/7.54.0'
+}
+"""Standard request header for sending requests to the API."""
+
+STATUS_URL_PARSE_REGEX = re.compile(r'(\d+)$')
+"""Regex for parsing the report ID from the status URL."""
+
+
+class RTVFailure(Exception):
+    """Exception raised when there is an error with the connector."""
+
+
+class RockTheVote:
+    """
+    `Args:`
+        partner_id: str
+            The RockTheVote partner ID for the RTV account
+        partner_api_key: str
+            The API Key for the partner
+    `Returns`:
+        RockTheVote class
+    """
+
+    def __init__(self, partner_id=None, partner_api_key=None):
+        self.partner_id = check_env.check('RTV_PARTNER_ID', partner_id)
+        self.partner_api_key = check_env.check('RTV_PARTNER_API_KEY', partner_api_key)
+
+        self.client = APIConnector('https://vr.rockthevote.com/api/v4', headers=REQUEST_HEADERS)
+
+    def create_registration_report(self, before=None, since=None):
+        """
+        Create a new registration report.
+
+        `Args:`
+            before: datetime.datetime
+                Date before which to return registrations for
+            since: datetime.datetime
+                Date for filtering registrations
```

Comment:
Mostly just acknowledging that passing in a `datetime` object is a new pattern for us. How do we feel about it?

Reply:
Also, if you pass both parameters is the report time bound within those two dates?

Reply:
I am changing the code to accept a string.. and then I'm going to pull in a `dateutil` library to help [parse](https://dateutil.readthedocs.io/en/stable/parser.html) the strings into a `datetime`.

Reply:
In my testing, it looks like specifying both results in registrations from between the two dates, yes.